flexmeasures/api/v3_0/assets.py

```diff
@@ -81,7 +140,50 @@ def index(self, account: Account):
         :status 403: INVALID_SENDER
         :status 422: UNPROCESSABLE_ENTITY
         """
-        return assets_schema.dump(account.generic_assets), 200
+
+        # find out which accounts are relevant
+        accounts = []
+        if all_accessible:
+            accounts = get_acessible_accounts()
+            if account is not None:
+                check_access(account, "read")
```

Comment:
This looks odd to me. What is the expected behaviour when both `all_accessible is True` and an `account_id` is passed?

Reply:
Hm, true ... what is off here is that we check access to `account`, but it is not explicitly included in `accounts`. If it is, it came via `all_accessible`.

The current logic actually explicitly ignores `account_id`  if `all_accessible` is `True`. But that is not documented well.

I will document this distinction better and then remove this unnecessary auth check.